src/main/java/com/icthh/xm/tmf/ms/communication/utils/LogUtil.java

```diff
@@ -0,0 +1,17 @@
+package com.icthh.xm.tmf.ms.communication.utils;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+
+@Slf4j
+@UtilityClass
+public class LogUtil {
+
+    public static void withLog(String tenant, String command, Runnable action, String logTemplate, Object... params) {
+        final StopWatch stopWatch = StopWatch.createStarted();
+        log.info("[{}] start: {} " + logTemplate, tenant, command, params);
+        action.run();
```

Comment:
you can add try - catch - throw logic here and use for all cases where log timing is needed

Reply:
see comment on the top of the page